File: games/tp/serv/room.ts
import { Table } from "./table";
import { Player } from "../../../lib/player";

/**
 * 房间
 */
export class Room {
    //房间id
    private _id: number;

    //桌子
    private _table: Map<number, Table>;

    //玩家集合
    private _players: Map<number, RoomPlayer>;
}

/**
 * 房间里的玩家
 */
export class RoomPlayer extends Player {

}

File: serv/games/tp/help/room.go
package help

import (
	"github.com/hanjingo/casino/serv/base"
)

type Room struct {
	base.Room
	Players map[int64]*Player //玩家集合
	Tables  map[int]*Table    //桌子集合
	Turns   map[uint64]*Turn  //回合集合
}

//初始化
func (room *Room) Init() {

}

//关闭
func (room *Room) Close() {

}

//结算
func (room *Room) Settle() {

}
